src/Umbraco.Web.UI.Client/src/packages/documents/documents/url/document-urls-data-resolver.ts

```diff
@@ -0,0 +1,94 @@
+import type { UmbDocumentUrlModel } from './repository/types.js';
+import { UmbControllerBase } from '@umbraco-cms/backoffice/class-api';
+import type { UmbControllerHost } from '@umbraco-cms/backoffice/controller-api';
+import { UmbArrayState } from '@umbraco-cms/backoffice/observable-api';
+import { UMB_PROPERTY_DATASET_CONTEXT } from '@umbraco-cms/backoffice/property';
+import type { UmbVariantId } from '@umbraco-cms/backoffice/variant';
+
+/**
+ * A controller for resolving data for document urls
+ * @exports
+ * @class UmbDocumentUrlsDataResolver
+ * @augments {UmbControllerBase}
+ */
+export class UmbDocumentUrlsDataResolver extends UmbControllerBase {
+	#appCulture?: string;
```

Comment:
Seems unfinished as this is never set? I assume we need to revisit when mergin in the Validation Context?